src/action_builder.js

```diff
@@ -621,6 +632,33 @@ module.exports = class ActionBuilder {
       this._showHints = false;
       this._links = [];
     }
+
+    // init deployers
+    const targets = { };
+    this._target.forEach((t) => {
+      if (t === 'auto') {
+        // get all deployers that are ready();
+        Object.entries(this._deployers).forEach(([name, deployer]) => {
+          if (deployer.ready()) {
+            targets[name] = deployer;
+          }
+        });
+      } else {
+        // deployer must be ready
+        const deployer = this._deployers[t];
+        if (!deployer) {
+          throw Error(`'No such target: ${t}`);
+        }
+        deployer.validate();
+        targets[t] = deployer;
+      }
+    });
```

Comment:
This would be cleaner as a `filter` sequence.

```suggestion
    const targets = Object.entries()
      .filter(([name]) => this._targets.includes(name) || this._targets.includes('auto'))
      .filter(([_, deployer]) => deployer.ready())
      .reduce((t, [name, deployer]) => {
        deployer.validate();
        t[name] = deployer;
        return t;
      }, {});
```

Reply:
not really. your `ready()`-filter would filter out the explicitely selected targets, which then won't get called for validate().